src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Compatibility.Common.targets

```diff
@@ -0,0 +1,36 @@
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. -->
+<Project>
+  <UsingTask TaskName="Microsoft.DotNet.PackageValidation.ValidatePackage" AssemblyFile="$(DotNetPackageValidationAssembly)" />
+
+  <Target Name="RunPackageValidation"
+          AfterTargets="Pack"
+          Condition="'$(IsPackable)' == 'true' and '$(EnablePackageValidation)' == 'true'">
+
+    <PropertyGroup Condition="'$(RoslynAssemblyPath)' == ''">
+      <RoslynAssemblyPath>$(RoslynTargetsPath)</RoslynAssemblyPath>
+      <RoslynAssemblyPath Condition="'$(MSBuildRuntimeType)' == 'Core'">$([System.IO.Path]::Combine('$(RoslynAssemblyPath)', bincore))</RoslynAssemblyPath>
```

Comment:
Can/should Roslyn own initialization of this property? IIRC Jared has talked about how the paths to our assemblies are subject to change in the future. For example the net472 assemblies may move into a `bin/` folder next to the `bincore/` folder that the .NET Core assemblies currently live in. Wondering what you think @chsienki.

Reply:
> Can/should Roslyn own initialization of this property?

yeah thats the idea for future. But this is the best we can do for now.

Reply:
Agreed that Roslyn should own this, but in the short term it seems fine to put this in the SDK. 

Even if 3rd parties start taking a dependency, then I think it should still work as this target is going to have to import the roslyn ones, meaning it'll flow through when we move it. I opened https://github.com/dotnet/roslyn/issues/55186 to track this.

Reply:
@RikkiGibson @chsienki are you OK with the property name `RoslynAssemblyPath` as it has been defined here, so that when Roslyn defines it we don't need to react?

Reply:
@ericstj Yeah, I guess my only nit would be make it Assemb**lies** path as there is more than one and it matches with the `RoslynTargetsPath`. Otherwise I'm not sure what else we would call it.